protocol/bc/asset.go

```diff
@@ -10,14 +10,14 @@ import (
 
 // AssetID is the Hash256 of the issuance script for the asset and the
 // initial block of the chain where it appears.
-type AssetID [32]byte
+type AssetID Byte32
 
-func (a AssetID) String() string                { return Hash(a).String() }
-func (a AssetID) MarshalText() ([]byte, error)  { return Hash(a).MarshalText() }
-func (a *AssetID) UnmarshalText(b []byte) error { return (*Hash)(a).UnmarshalText(b) }
-func (a *AssetID) UnmarshalJSON(b []byte) error { return (*Hash)(a).UnmarshalJSON(b) }
-func (a AssetID) Value() (driver.Value, error)  { return Hash(a).Value() }
-func (a *AssetID) Scan(b interface{}) error     { return (*Hash)(a).Scan(b) }
+func (a AssetID) String() string                { return Byte32(a).String() }
```

Comment:
If we wanted to keep `type AssetID [32]byte` for now, these functions could look like

```go
func (a AssetID) String() string                { return NewHash(a).String() }
```

(assuming the existence of a NewHash function)